apps/user_ldap/lib/Group_LDAP.php

```diff
@@ -139,30 +146,37 @@ public function inGroup($uid, $gid) {
 		}
 
 		//extra work if we don't get back user DNs
-		if (strtolower($this->access->connection->ldapGroupMemberAssocAttr) === 'memberuid') {
-			$dns = [];
-			$filterParts = [];
-			$bytes = 0;
-			foreach ($members as $mid) {
-				$filter = str_replace('%uid', $mid, $this->access->connection->ldapLoginFilter);
-				$filterParts[] = $filter;
-				$bytes += strlen($filter);
-				if ($bytes >= 9000000) {
-					// AD has a default input buffer of 10 MB, we do not want
-					// to take even the chance to exceed it
+		switch ($this->ldapGroupMemberAssocAttr) {
+			case 'memberuid':
+			case 'zimbramailforwardingaddress':
+				$dns = [];
+				$filterParts = [];
+				$bytes = 0;
+				foreach ($members as $mid) {
+					if ($this->ldapGroupMemberAssocAttr === 'zimbramailforwardingaddress') {
+						$parts = explode('@', $mid); //making sure we get only the uid
+						$mid = $parts[0];
+					}
+					$filter = str_replace('%uid', $mid, $this->access->connection->ldapLoginFilter);
+					$filterParts[] = $filter;
+					$bytes += strlen($filter);
+					if ($bytes >= 9000000) {
+						// AD has a default input buffer of 10 MB, we do not want
+						// to take even the chance to exceed it
+						$filter = $this->access->combineFilterWithOr($filterParts);
+						$bytes = 0;
+						$filterParts = [];
+						$users = $this->access->fetchListOfUsers($filter, 'dn', count($filterParts));
```

Comment:
mh, just  noticing that `$filterParts` is reset in the line above, it will always be 0. That was already the case with the old code. The reseting lines should be moved below. It is not making a dramatic difference, but make it possible that the answer is returned faster.

```suggestion
						$users = $this->access->fetchListOfUsers($filter, 'dn', count($filterParts));
						$bytes = 0;
						$filterParts = [];
```

Reply:
yes, makes sense to me!